app/views/ChooseProvider.js

```diff
@@ -60,22 +56,11 @@ class ChooseProviderScreen extends Component {
     return true;
   };
 
-  componentDidMount() {
+  async componentDidMount() {
     BackHandler.addEventListener('hardwareBackPress', this.handleBackPress);
-    this.fetchAuthoritiesList();
-
-    // Update user settings state from async storage
-    GetStoreData(AUTHORITY_SOURCE_SETTINGS, false).then(result => {
-      if (result !== null) {
-        console.log('Retrieving settings from async storage:');
-        console.log(result);
-        this.setState({
-          selectedAuthorities: result,
-        });
-      } else {
-        console.log('No stored authority settings.');
-      }
-    });
+    await this.fetchAuthoritiesList(this.state.isAuthorityFilterActive);
+    await this.fetchUserAuthorities();
+    await this.fetchAutoSubcribeStatus();
```

Comment:
this reads super well, thanks!

+100 for not lumping 4 layers of abstraction into this method.